frontend/src/App.jsx

```diff
@@ -1,22 +1,58 @@
-import './App.css'
 import HomePage from "./components/homePage/HomePage.jsx"
-
+import ForYouPage from "./components/forYouPage/ForYouPage.jsx"
+import NearbyPage from "./components/nearbyPage/NearbyPage.jsx"
+import EventAttendingPage from "./components/eventAttendingPage/EventAttendingPage.jsx"
+import FavoriteEventsPage from "./components/favoriteEventsPage/FavoriteEventsPage.jsx"
+import InitialPage from "./components/initialPage/InitialPage.jsx"
+import SignupPage from "./components/signUpPage/SignUpPage.jsx"
 import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
 
+{/* testing for user auth */}
+import LoginForm from "./components/LoginForm/LoginForm.jsx"
+import SignupForm from "./components/SignupForm/SignupForm.jsx"
+import { useState, useEffect } from 'react';
+import { UserContext } from './UserContext';
+
+
 function App() {
 
+  const [user, setUser] = useState(() => {
+    // Retrieve the user data from storage or set it to null if not found
+    const storedUser = localStorage.getItem('user');
+    // return storedUser ? JSON.parse(storedUser) : null;
+    return null
+  });
```

Comment:
Does this actually work? You're initializing `user` to be a function that returns `storedUser`. I'm assuming you want it to be the actual user, not a function.
So initially, it should be `const [user, setUser] = useState(null);`

If you want to try to see if you already stored the user in local storage and fetching it on initialization, you can do
```javascript
useEffect(() => {
  const storedUser = localStorage.getItem('user');
  setUser(storedUser);
}, []);
```

Reply:
Or even better

```
const [user, setUser] = useState(localStorage.getItem('user') ?? null);
```

Reply:
Thank you. I was having problems with that

Reply:
To do

Reply:
Done